### PR TITLE
[IconFetcher] Fix ABI compatibility with upstream rofi

### LINKED ISSUE #96: [REQUEST] ABI compatibility with Rofi?
### Before opening a feature request

- [X] I checked the *next* branch to see if the feature has already been implemented
- [X] I searched existing reports to see if it is already requested.

### What is the user problem or growth opportunity you want to see solved?

Plugins built for normal Rofi are currently ABI-incompatible with plugins built for this (Wayland) Rofi; however, this is not made obvious to the user (attempting to load a Rofi plugin using Wayland Rofi simply results in UB, manifesting as an unhelpful assertion failure), nor is it easy for plugin authors to support both normal and Wayland Rofi.

I would suggest one of two things:
1. undo the breaking changes made to Rofi’s plugin ABI (most importantly, the extra parameter in `_mode_get_icon`, but `rofi_icon_fetcher_query[_advanced]` may also be changed), or if that is not possible
2. bump the ABI version number from 7 to something like 1007, to ensure that the error is made obvious to the user.

### How do you know that this problem exists today? Why is this important?

Some people use Wayland Rofi for practical use, as it is in the AUR.

My interest in this comes from wanting to support Wayland Rofi in my Rust library for writing Rofi plugins: https://github.com/SabrinaJewson/rofi-mode.rs/issues/8.

### Who will benefit from it?

People who write and use plugins for Wayland Rofi.

### Rofi version (rofi -v)

Latest commit from repositories, as of 2023-10-31

### Configuration

N/A

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by moetayuko**
Fixes #96

cc @SabrinaJewson

### CODE REVIEW

**Comment by moetayuko** [comment]
@q234rty plz test if any regressions.

**Comment by q234rty** [comment]
Seems to work fine in my case.

File: source/theme.c [hunk 1]
```diff
@@ -1114,14 +1114,21 @@ static gboolean rofi_theme_get_image_inside(Property *p, const widget *widget,
       default:
         break;
       }
+#if 0
```
Comment (lbonn) [inline source/theme.c:1117]: Do we still need this commented out code?

Comment (moetayuko) [reply source/theme.c:1117]: I just left it for future reference. If you want I can remove it.

**Comment by lbonn** [comment]
@MoetaYuko I have reworked the changes there #102, please have a look

**Comment by lbonn** [comment]
Closed after #102

**Pull request closed without merging**